ethers-ffi/src/lib.rs

```diff
@@ -72,16 +72,20 @@ pub struct SignedTransaction {
 pub struct CSignedTransaction {
     signature: *const c_char,
 }
-
 #[no_mangle]
-pub extern "C" fn generate_mnemonic() -> CMnemonicAndAddress {
+fn generate_mnemonic_rust() -> MnemonicAndAddress {
```

Comment:
abstracted this to a separate function so that we don't have to deal with converting between c representation for java (where as we do for c in iOS)